internal/controller/vrg_kubeobjects.go

```diff
@@ -643,7 +643,7 @@ func (v *VRGInstance) kubeObjectsRecoveryStartOrResume(
 
 		if rg.IsHook {
 			if err := v.executeHook(rg.Hook, log1); err != nil {
-				break
+				return fmt.Errorf("check hook execution failed during restore %s: %v", rg.Hook.Name, err)
```

Comment:
Do we want to wrap the error with %w? Is it helpful for code calling this, maybe checking for specific error with errors.Is()?

Reply:
I don't think so. We don't want the caller to interpret the errors. We can make that change later if we think these errors need to be inspected in the caller.